slu/slu/src/controller/processors.py

```diff
@@ -23,7 +23,30 @@ def get_plugins(self, purpose, **kwargs) -> List[Plugin]:
             input_column=const.ALTERNATIVES,
             debug=self.debug,
         )
+
+        xlmr_clf = plugins.XLMRMultiClass(
+            dest="output.intents",
+            model_dir=self.config.get_model_dir(const.CLASSIFICATION),
+            score_round_off=5,
+            purpose=purpose,
+            use_cuda=purpose != const.PRODUCTION,
+            data_column=const.ALTERNATIVES,
+            label_column=const.TAG,
+            args_map=self.config.get_model_args(const.CLASSIFICATION, purpose, epochs=kwargs.get(const.EPOCHS)),
+            debug=self.debug,
+        )
+
+        oos_filter = OOSFilterPlugin(
+            dest="output.intents",
+            threshold=self.config.get_model_confidence_threshold(const.CLASSIFICATION),
+            replace_output=True,
+            guards=[lambda i, o: purpose != const.PRODUCTION],
+        )
         
+        retrain_original_intent = plugins.RetainOriginalIntentPlugin(
+            debug=self.debug
+        )
```

Comment:
is there a reason why `RetrainOriginalIntentPlugin` needs to come after `OOSFilterPlugin`.
can't it be directly used after `XLMRMultiClass` ?

Reply:
if we put it before then I think we won't be using the context of oos then as most model won't predict oos directly and even if they do predict oos directly this oosfilter plugin won't change that (as it starts to work only based on the threshold). 
Now since original intent is to compare with ground truth and we would always be tagging an oos datapoint as under of the oos layers only in ground truth, so this comparison would make sense I think, do you agree with this?

Reply:
oh yeah makes sense.